ivy/functional/backends/tensorflow/experimental/statistical.py

```diff
@@ -45,3 +45,38 @@ def unravel_index(
 ) -> Union[tf.Tensor, tf.Variable]:
     ret = tf.unravel_index(indices, shape)
     return [tf.constant(ret[i]) for i in range(0, len(ret))]
+
+
+def nanquantile(
+    a: Union[tf.Tensor, tf.Variable],
+    q: Union[int, float],
+    /,
+    *,
+    axis: Optional[Union[int, Tuple[int]]] = None,
+    keepdims: Optional[bool] = False,
+    interpolation: Optional[str] = None,
+    out: Optional[Union[tf.Tensor, tf.Variable]] = None,
+) -> Union[tf.Tensor, tf.Variable]:
+    count = 0
+    ret = []
+    res = 0
+    for i in a:
+        for j in i:
```

Comment:
when you go into the nested loop, the value becomes a scalar and you cannot iterate through a scalar, also for the implementation of this I am referring you to a link from where you can understand what `nanquantile()` actually does and use the formula used there for the it's implementation.
https://www.statisticshowto.com/quantile-definition-find-easy-steps/